libr/asm/arch/gb/gbasm.c

```diff
@@ -16,59 +19,64 @@ static int gb_reg_idx (char r) {
 
 static bool gb_parse_cb1 (ut8 *buf, const int minlen, char *buf_asm, ut8 base) {
 	int i;
-	if (strlen (buf_asm) < minlen)
+	if (strlen (buf_asm) < minlen || minlen < 1) {
 		return false;
+	}
 	buf[0] = base;
-	i = strlen (&buf_asm[minlen - 1]);
-	r_str_replace_in (&buf_asm[minlen - 1], (ut32)i, "[ ", "[", true);
-	r_str_replace_in (&buf_asm[minlen - 1], (ut32)i, " ]", "]", true);
+	i = strlen (buf_asm + minlen - 1);
+	r_str_replace_in (buf_asm + minlen - 1, (ut32)i, "[ ", "[", true);
+	r_str_replace_in (buf_asm + minlen - 1, (ut32)i, " ]", "]", true);
 	r_str_do_until_token (str_op, buf_asm, ' ');
 	i = gb_reg_idx (buf_asm[minlen-1]);
```

Comment:
You probably want to use new way here too.

Reply:
thats not a memory dereference &